src/mixxxapplication.cpp

```diff
@@ -207,10 +217,43 @@ bool MixxxApplication::notify(QObject* target, QEvent* event) {
 #endif // QT_VERSION < QT_VERSION_CHECK(5, 0, 0)
 #endif // Q_OS_MAC
 
+bool MixxxApplication::notify(QObject* target, QEvent* event) {
+    // All touch events are translated in two simultaneous events one for
+    // target QWidgetWindow and one for the target QWidget
+    // A second touch becomes a mouse move without additional press and release
+    // events
+    switch (event->type()) {
+    case QEvent::MouseButtonPress: {
+        QMouseEventEditable* mouseEvent = static_cast<QMouseEventEditable*>(event);
+        if (mouseEvent->source() == Qt::MouseEventSynthesizedByQt &&
+                mouseEvent->button() == Qt::LeftButton &&
+                touchIsRightButton()) {
+            mouseEvent->setButton(Qt::RightButton);
+            m_rightPessedButtons++;
+            DEBUG_ASSERT(m_rightPessedButtons <= 2);
+        }
+        break;
+    }
+    case QEvent::MouseButtonRelease: {
+        QMouseEventEditable* mouseEvent = static_cast<QMouseEventEditable*>(event);
+        if (mouseEvent->source() == Qt::MouseEventSynthesizedByQt &&
+                mouseEvent->button() == Qt::LeftButton &&
+                m_rightPessedButtons > 0) {
```

Comment:
VERIFY_OR_DEBUG_ASSERT(m_rightPessedButtons > 0) and early break otherwise?

Using a debug assert for increasing the counter but not for decreasing it would be inconsequent. Both parts of the code should be symmetrical.

Reply:
m_rightPessedButtons is zero when there is no right button pressed. This is an expected case. 
The assertion in the other path verifies the assumption that QT synthesizes only one click at a time.   I will add a comment.